src/main/java/core/basesyntax/serviceimpl/OperationStrategyImpl.java

```
@@ -0,0 +1,24 @@
+package core.basesyntax.serviceimpl;
+
+import core.basesyntax.common.FruitTransaction;
+import core.basesyntax.service.OperationStrategy;
+import core.basesyntax.strategy.OperationHandler;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> operationHandlerMap;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation,
+            OperationHandler> operationHandlerMap) {
+        this.operationHandlerMap = operationHandlerMap;
+    }
+
+    @Override
+    public OperationHandler getOperationHandler(FruitTransaction.Operation operation) {
+        OperationHandler operationHandler = operationHandlerMap.get(operation);
+        if (operationHandler == null) {
+            throw new IllegalArgumentException("Unknown operation: " + operation);
```

When no handler is found the code throws `IllegalArgumentException` with a brief message (lines 19–20). Improve this to provide more context (include the operation code and optionally list supported operations) and consider using a RuntimeException with an explicit message per the project's error-handling guidance. The checklist expects informative RuntimeExceptions rather than letting ambiguous exceptions propagate. Quote: "Don't ignore exceptions in try-catch blocks ... rethrow a RuntimeException with an **informative** message and exception object."